source/common/certificate_provider/certificate_provider_manager_impl.h

```diff
@@ -0,0 +1,34 @@
+#pragma once
+
+#include <string>
+
+#include "envoy/api/api.h"
+#include "envoy/certificate_provider/certificate_provider_manager.h"
+#include "envoy/config/core/v3/extension.pb.h"
+
+#include "absl/container/flat_hash_map.h"
+
+namespace Envoy {
+namespace CertificateProvider {
+
+/**
+ * A manager for certificate provider instances.
+ */
+class CertificateProviderManagerImpl : public CertificateProviderManager {
+public:
+  CertificateProviderManagerImpl(Api::Api& api);
+
+  void addCertificateProvider(std::string name,
+                              const envoy::config::core::v3::TypedExtensionConfig& config) override;
+
+  CertificateProviderSharedPtr getCertificateProvider(std::string name) override;
+
+private:
+  absl::flat_hash_map<std::string, CertificateProviderSharedPtr> certificate_provider_instances_;
+  Api::Api& api_;
+};
+
+using CertificateProviderManagerPtr = std::unique_ptr<CertificateProviderManager>;
```

Comment:
duplicate: (already in envoy/certificate_provider/certificate_provider_manager.h)

Reply:
got it